sdk/messaging/azservicebus/receiver.go

```diff
@@ -85,7 +81,7 @@ type ReceiverOptions struct {
 	SubQueue SubQueue
 }
 
-const defaultLinkRxBuffer = 2048
+const defaultLinkRxBuffer = 10000
```

Comment:
Make sure this isn't larger than the incoming window.  While it doesn't really matter today, it might in the future.

Reply:
It also seems quite large.  Do you really think we need to change it?  How do we measure the impact at any value?

Reply:
> Make sure this isn't larger than the incoming window. While it doesn't really matter today, it might in the future.

How can I determine the incoming window value? Or maybe the other way - shouldn't the incoming window value be calculated based on my max link credit?

> It also seems quite large. Do you really think we need to change it? How do we measure the impact at any value?

I'd much rather not set it at all and have it be dynamic, based on whatever the outstanding issued credit is. In the meantime I have to pick some ceiling. What would you suggest?

Reply:
From this conversation, I think it's misleading to call the option `ReceiverOptions.MaxCredit`.  It's not the maximum amount of credit on the link (that's essentially 2^32).  It's the maximum amount of prefetched messages.  I think the field needs to be renamed.

Reply:
(from other discussion, I think we might end up eliminating this field altogether)